master/security/adopt-zero-trust.md

```diff
@@ -0,0 +1,314 @@
+
+#### How Calico and Istio implement Zero Trust Network requirements
+
+Calico works in concert with the Istio service mesh to implement all you need to build a Zero Trust Network in your Kubernetes cluster.
+
+##### Multiple enforcement points
+
+When operating with Istio, incoming requests to your workloads traverse two distinct enforcement points:
+
+1. The host Linux kernel. Calico policy is enforced in the Linux kernel using iptables at L3-L4.
+1. The Envoy proxy. Calico policy is enforced in the Envoy proxy at L3-7, with requests being cryptographically authenticated. A lightweight policy decision sidecar called Dikastes assists Envoy in this enforcement.
+
+These multiple enforcement points establish the identity of the remote endpoint based on multiple criteria (Requirement 2). The host Linux kernel enforcement protects your workloads even if the workload pod is compromised and the Envoy proxy bypassed (Requirement 4).
+
+##### Calico policy store
+
+The policies in the Calico data store encode the whitelist of allowed flows (Requirement 3).
+
+Calico network policy is designed to be flexible to fit many different security paradigms, so it can express, for example, both Zero Trust Network-style whitelists as well as legacy paradigms like zones. You can even layer both of these approaches on top of one another without creating a maintenance mess by composing multiple policy documents.
+
+The How To section of this document explains how to write policy specifically in the style of Zero Trust Networks. Conceptually, you will begin by denying all network flows by default, then add rules that allow the specific expected flows that make up your application. When you finish, only legitimate application flows are allowed and all others are denied.
+
+##### Calico control plane
+
+The Calico control plane handles distributing all the policy information from the Calico data store to each enforcement point, ensuring that all network connections are subject to enforcement (Requirement 4). It translates the high-level declarative policy into the detailed enforcement attributes that change as applications scale up down to meet demand, and evolve as developers modify them.
+
+##### Istio Citadel Identity System
+
+In Calico and Istio, workload identities are based on Kubernetes Service Accounts. An Istio component called Citadel handles minting cryptographic keys for each Service Account to prove its identity on the network (Requirement 2) and encrypt traffic (Requirement 5). This allows the Zero Trust Network to be resilient even if attackers compromise network infrastructure like routers or links.
+
+### Before you begin...
+
+[Enable Application Layer Policy]({{site.baseurl}}/{{page.version}}/getting-started/kubernetes/installation/app-layer-policy)
```

Comment:
Actually, move this link to line 116.

I didn't remember that we have that as an explicit step in the how-to.

Reply:
Fixed.